src/resolvers/subjectResolvers.ts

```diff
@@ -10,25 +10,44 @@ import { ISubjectPageData } from '@ndla/types-frontpage-api';
 import { fetchLK20CompetenceGoalSet, fetchSubjectPage } from '../api';
 
 import { filterMissingArticles } from '../utils/articleHelpers';
+import { Subject } from '../api/taxonomyApi';
 
 export const Query = {
   async subject(
     _: any,
     { id }: QueryToSubjectArgs,
     context: ContextWithLoaders,
-  ): Promise<GQLSubject | undefined> {
-    const data = await context.loaders.subjectsLoader.load('all');
-    return data.subjects
-      .filter(s => (s.metadata ? s.metadata.visible : true))
-      .find(subject => subject.id === id);
+  ): Promise<Subject> {
```

Comment:
Er det ikkje naturlig å returnere GQLSubject her? I schema.ts så returneres jo Subject, som GQLSubject baseres på. Dette virker som går på tvers av det vi gjør alle andre plasser?

Reply:
En query returnerer jo ikke nødvendigvis en fullverdig `GQLType`, det kan kun gjøres ved å kombinere en query med custom resolvers. Hvis vi for eksempel hadde lagt inn `grepCodesFromUdir` som et påkrevd felt på `GQLSubject` ville jo queryen ikke klare å tilfredsstille `GQLSubject`.

Reply:
Det er sant. Det har eg ikkje tenkt på. Men Subject og GQLSubject er jo uavhengige typer. Må du manuellt passe på at dei er like nok?

Reply:
Eventuelle felt fra resolvers må jo per def være frivillige fordi dei ikkje hentes fra kilden (taxonomy-api). Då kan jo returtypen være GQLSubjects fordi den innehelde påkrevde felt?

Reply:
Inntil videre, ja. Tror derimot det er mulig å få det til, men ikke uten at vi kommer oss over til `strict: true` i tsconfig. Og det er det en stund til vi kan gjøre.